### DISCUSSION TITLE
Can I have GCM save my gitconfig changes to a different file?

### DISCUSSION THREAD

**Discussion by sheldonhull** [git-ecosystem/git-credential-manager#1404] [Q&A] [answered]
- I have my `.gitconfig` version controlled and templated.
- Using SSO with GCM changes my `.gitconfig`.
- Is there a way I can have GCM _not_ change my gitconfig while still preserving the SSO setup that gcm does interactively with the web browser?

## Using include

I tried to setup:

```config
# $HOME/.gitconfig
[include]
    path = .gitconfig.gcm
```

and then in `$HOME/.gitconfig.gcm`

```config
# File for git credential manager to update without impacting my global config
[credential "azrepos:org/ORG"]
    username = myssoemail
```

When I restarted code and it didn't see to honor this and immediately prompted for SSO and edited my `$HOME/.gitconfig`. I'm not sure if it's an issue with me and git configs or how GCM works :-)

**Comment by ldennington** [comment, marked as answer]
@sheldonhull - unfortunately GCM does not currently recognize config includes and will thus always update `$HOME/.gitconfig`. This is something we'd like to add, but it would involve a pretty complex change (essentially re-writing Git's config parsing logic). We're tracking with [this issue](https://github.com/git-ecosystem/git-credential-manager/issues/1337).

**Reply by sheldonhull** [reply]
Sounds great. That helps. I’ll just move my settings into an includes file and should be solved.
